workspaces/services/guacamole-azure-dsvm/terraform/dsvm.tf

```diff
@@ -0,0 +1,128 @@
+resource "azurerm_network_interface" "internal" {
+  name                = "internal-nic-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+
+  ip_configuration {
+    name                          = "primary"
+    subnet_id                     = data.azurerm_subnet.dsvms.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_security_group" "dsvmnsg" {
+  name                = "nsg-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "allow-rdp-inbound-from-vnet-dsvm" {
+  name                        = "RDP"
+  priority                    = 800
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 3389
+  source_address_prefix       = "VirtualNetwork"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  resource_group_name         = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "deny-all-inbound-override-dsvm" {
+  access                      = "Deny"
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+  direction                   = "Inbound"
+  name                        = "deny-inbound-override"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  priority                    = 900
+  protocol                    = "*"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_address_prefix       = "*"
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "deny-outbound-override-dsvm" {
```

Comment:
This will block all outbound access? Traffic needs to be allowed out - to the Internet in the NSG - but routed using the default route on the subnet to direct through the firewall which then restricts the traffic. As above would use the existing nsg for the workspace and services subnet.

Reply:
removed and moved to the ws nsg and subnet